docs/source/deployment/airflow.md

```diff
@@ -0,0 +1,173 @@
+
+### Prerequisites
+
+To follow this tutorial, ensure you have the following:
+
+* The [Astro CLI installed](https://docs.astronomer.io/astro/install-cli)
+* A container service like [Docker Desktop](https://docs.docker.com/get-docker/) (v18.09 or higher)
+* `kedro>=0.19` installed
+* [`kedro-airflow>=0.8`](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-airflow) installed. We will use this plugin to convert the Kedro pipeline into an Airflow DAG.
+
+### Create, prepare and package example Kedro project
+
+In this section, you will create a new Kedro project equipped with an example pipeline designed to solve a typical data science task: predicting spaceflights prices. You will need to customise this project to ensure compatibility with Airflow, which includes enriching the Kedro `DataCatalog` with datasets previously stored only in memory and simplifying logging through custom settings. Following these modifications, you will package the project for installation in an Airflow Docker container and generate an Airflow DAG that mirrors our Kedro pipeline.
+
+1. To create a new Kedro project, select the `example=yes` option to include example code. Additionally, to implement custom logging, select `tools=log`. Proceed with the default project name, but feel free to add any other tools as desired:
+
+    ```shell
+    kedro new --example=yes --name=new-kedro-project --tools=log
+    ```
+
+2. Navigate to your project's directory, create a new `conf/airflow` directory for Airflow-specific configurations, and copy the `catalog.yml` file from `conf/base` to `conf/airflow`. This setup allows you to customise the `DataCatalog` for use with Airflow:
+
+    ```shell
+    cd new-kedro-project
+    mkdir conf/airflow
+    cp conf/base/catalog.yml conf/airflow/catalog.yml
+    ```
+
+3. Open `conf/airflow/catalog.yml` to see the list of datasets used in the project. Note that additional intermediate datasets (`X_train`, `X_test`, `y_train`, `y_test`) are stored only in memory. You can locate these in the pipeline description under `/src/new_kedro_project/pipelines/data_science/pipeline.py`. To ensure these datasets are preserved and accessible across different tasks in Airflow, we need to include them in our `DataCatalog`. Instead of repeating similar code for each dataset, you can use [Dataset Factories](https://docs.kedro.org/en/stable/data/kedro_dataset_factories.html), a special syntax that allows defining a catch-all pattern to overwrite the default MemoryDataset creation. Add this code to the end of the file:
+
+```yaml
+{IntermediateDataset}:
+  type: pandas.CSVDataset
+  filepath: data/02_intermediate/{IntermediateDataset}.csv
+```
```

Comment:
nit: Not terribly happy about this `CamelCase` for dataset replacements, @ankatiyar wdyt?

````suggestion
```yaml
{intermediate_dataset}:
  type: pandas.CSVDataset
  filepath: data/02_intermediate/{intermediate_dataset}.csv
```
````

Reply:
I agree, I think we've got lower case pattern names in dataset factory docs as well. Another thing I would like to point out is there is a bit of an inconsistency with the naming of the default dataset - https://github.com/kedro-org/kedro/issues/3720
Right now it'll have to be ranked higher that `{default}` which is passed by runners - so the name should begin with A/B/C it's weird but I don't think this will work with the current state of things.

Reply:
updated to `{base_dataset}`